R/formatter.r

```diff
@@ -372,27 +388,28 @@ precision <- function(x) {
   10^floor(log10(span))
 }
 
-#' Add units to the labels
-#'
-#' @param unit The units to append
-#' @param scale A scaling factor. Useful if the underlying data is on another scale
-#' @param sep The separator between the number and the label
-#' @param ... Arguments passed on to [format()]
+
+#' `unit_format` add units to the values.
 #' @export
+#' @rdname number_format
+#' @param unit The units to append.
+#' @param sep The separator between the number and the unit label.
 #' @examples
+#' # unit_format()
 #' # labels in kilometer when the raw data are in meter
 #' km <- unit_format(unit = "km", scale = 1e-3, digits = 2)
 #' km(runif(10) * 1e3)
 #'
 #' # labels in hectares, raw data in square meters
 #' ha <- unit_format(unit = "ha", scale = 1e-4)
 #' km(runif(10) * 1e5)
-#' @seealso [comma()]
-unit_format <- function(unit = "m", scale = 1, sep = " ", ...) {
-  force_all(unit, scale, sep, ...)
-  function(x) {
-    paste(comma(x * scale, ...), unit, sep = sep)
-  }
+#'
+unit_format <- function(unit = "m", sep = " ", suffix = paste0(sep, unit), ...) {
```

Comment:
I think it's still worth including scale here, just to be explicit.

Reply:
done